internal/encoded.go

```diff
@@ -25,19 +25,26 @@
 package internal
 
 import (
+	"bytes"
 	"encoding/base64"
 	"encoding/json"
 	"errors"
 	"fmt"
 	"reflect"
+	"strings"
 
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gogo/protobuf/proto"
 	commonpb "go.temporal.io/api/common/v1"
 )
 
 const (
-	metadataEncoding     = "encoding"
-	metadataEncodingRaw  = "raw"
-	metadataEncodingJSON = "json"
+	metadataEncoding          = "encoding"
+	metadataEncodingRaw       = "binary/raw"
+	metadataEncodingJSON      = "json/plain"
+	metadataEncodingNil       = "binary/null"
+	metadataEncodingProtoJSON = "json/protobuf"
+	metadataEncodingProto     = "binary/protobuf"
```

Comment:
Idea behind these names that `json/*` can be directly displayed in UI and basically human readable. `binary/` is vice versa.